drivers/include/periph/spi.h

```diff
@@ -329,7 +329,7 @@ typedef struct {
  * @retval  0           success
  * @retval  <0          error
  */
-int spi_init_with_gpio_mode(spi_t bus, spi_gpio_mode_t mode);
+int spi_init_with_gpio_mode(spi_t bus, const spi_gpio_mode_t* mode);
```

Comment:
The doxygen documentation should be updated as well (`Pointer to a struct, etc`)